### PR TITLE
fixed rcl_wait return error when timer cancelled

### PR DESCRIPTION
**Pull request by ladianchad**
when timer canclled, `rcl_timer_get_time_until_next_call` return `RCL_RET_TIMER_CANCELED`
rcl_wait return with "error not set" like below ( we checked `rcl_timer_is_canceled` but timer is atomic, we can miss this)

```bash
[planner_server-3] terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
[planner_server-3]   what():  rcl_wait() failed: error not set
```
so, I changed check timer canceled by `rcl_timer_get_time_until_next_call`'s return value.

Would these solutions be appropriate?

### CODE REVIEW

**Comment by ladianchad** [comment]
https://github.com/ros2/rclcpp/issues/1977
this bug also mentioned at this issue

**Review by fujitatomoya — APPROVED** [review]
lgtm!

**Comment by fujitatomoya** [comment]
@ladianchad thanks for the contribution, this looks good to me. can you address DCO error?

**Comment by fujitatomoya** [comment]
@iuhilnehc-ynos requesting another review on this.

CC: @ivanpauno

**Comment by ladianchad** [comment]
> @ladianchad thanks for the contribution, this looks good to me. can you address DCO error?

yes, I fixed DCO error and rebased my unnecessary commits

**Comment by iuhilnehc-ynos** [comment]
LGTM. The default branch to push is `rolling`, then we can backport it to `humbel`.

**Comment by ladianchad** [comment]
oh. thanks i changed pr branch to humble

File: rcl/src/rcl/wait.c [hunk 1]
```diff
@@ -558,19 +558,15 @@ rcl_wait(rcl_wait_set_t * wait_set, int64_t timeout)
...
-      }
-      if (is_canceled) {
-        wait_set->timers[i] = NULL;
-        continue;
-      }
       // use timer time to to set the rmw_wait timeout
       // TODO(sloretz) fix spurious wake-ups on ROS_TIME timers with ROS_TIME enabled
       int64_t timer_timeout = INT64_MAX;
-      ret = rcl_timer_get_time_until_next_call(wait_set->timers[i], &timer_timeout);
+
```
Comment (ivanpauno) [inline rcl/src/rcl/wait.c:564]:
```suggestion
```

**Review by ivanpauno — APPROVED** [review]
LGTM!

**Comment by fujitatomoya** [comment]
@ladianchad can you address @ivanpauno comment ? then i will start CI.

**Comment by ladianchad** [comment]
> @ladianchad can you address @ivanpauno comment ? then i will start CI.

I applied review in [1554188](https://github.com/ros2/rcl/pull/1003/commits/1554188cf02e0f0b640d0e4ceff61c96773cea6f) :)

**Comment by fujitatomoya** [comment]
@ladianchad thanks for the quick response.

CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17174/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11717/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17598/)

**Comment by ladianchad** [comment]
Linux Build Status is unstable, is it ok?? I don't quite understand ROS CI yet.

**Comment by fujitatomoya** [comment]
I do not think that is related.

CI(Retry Linux):

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17176/)

**Comment by ladianchad** [comment]
Thanks!

**Comment by fujitatomoya** [comment]
@Mergifyio backport humble galactic foxy

**Comment by ErikOrjehag** [comment]
Thank you for fixing this one! :D

**Pull request merged**